packages/flutter/test/widgets/selectable_region_test.dart

```diff
@@ -310,6 +310,36 @@ void main() {
       expect(edgeEvent.globalPosition, const Offset(200.0, 50.0));
     });
 
+  testWidgets(
+    "touch long press cancel doesn't send ClearSelectionEvent",
+    (WidgetTester tester) async {
+      final UniqueKey spy = UniqueKey();
+      await tester.pumpWidget(MaterialApp(
+        home: SelectableRegion(
+          focusNode: FocusNode(),
+          selectionControls: materialTextSelectionControls,
+          child: SelectionSpy(key: spy),
+        ),
+      ));
+      await tester.pumpAndSettle();
+
+      final RenderSelectionSpy renderSelectionSpy =
+          tester.renderObject<RenderSelectionSpy>(find.byKey(spy));
+      renderSelectionSpy.events.clear();
+      final TestGesture gesture =
+          await tester.startGesture(const Offset(200.0, 200.0));
+
+      addTearDown(gesture.removePointer);
+
+      await tester.pump(const Duration(milliseconds: 500));
+      await gesture.cancel();
```

Comment:
This seems like a reasonable way of testing the cancel. I think we should also test the specific situation of the cancel in a scrollable as a regression test. Something like long press to select a word in a scrollable. Then try scrolling. The selection should be retained after the scroll.

Also add a regression test comment at the top of the test. Something like
`// Regression test for https://github.com/flutter/flutter/issues/128765`